File: TodoApi/Models/Todo.cs
namespace TodoApi.Models;

public class Todo
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public bool IsComplete { get; set; }
    public string? Description { get; set; }
    public int user { get; set; } = 1;
}

File: react-todo-client/src/types/todo.ts
export interface Todo{
    id: number;
    title: string;
    isComplete: boolean;
    description: string | null;
    userId: number;
}

export interface User {
    id: number;
    name: string;
    age: number;
    email: string;
}

export type CreateTodoDto = {
  title: string;
  description?: string;
  userId: number;
};

export type UpdateTodoDto = {
    title: string;
    description?: string;
    isComplete: boolean;
    userId: number;
}